src/connectedk8s/azext_connectedk8s/custom.py

```diff
@@ -801,7 +859,20 @@ def update_agents(cmd, client, resource_group_name, cluster_name, https_proxy=""
 
     proxy_cert = proxy_cert.replace('\\', r'\\\\')
 
-    if https_proxy == "" and http_proxy == "" and no_proxy == "" and proxy_cert == "" and not disable_proxy and not auto_upgrade:
+    # Prompt if private link is getting enabled
+    if enable_private_link == "true":
+        if os.getenv('SKIP_PROMPT') != "true":
+            if not prompt_y_n("Enabling private link will disable 'cluster-connect' and 'custom-location' features. Are you sure you want to continue?"):
+                return
+
+    # Set preview client if private link properties are provided.
+    if enable_private_link:
+        client = cf_connected_cluster_prev_2021_04_01(cmd.cli_ctx, None)
+
```

Comment:
should we do get first here to check if private link was previously enabled on the cluster? If we expect user to pass all args every time then this should be fine.. but does portal takes care of passing all params during patch?

Reply:
I don't think a GET call is required. The preview SDK has to be used when user is making private link related operations during a cmd execution. For all other cases, GA SDK will be used.
The portal experience for PATCH is currently not there (except tags), But when we introduce it, the experience should be same as portal PUT operation(redirect to CLI)